README.md

```diff
@@ -325,6 +325,36 @@ To reimport just one entity, run:
 bundle exec rails dfe:analytics:import_entity[entity_name]
 ```
 
+
+## Event debugging
+
+If you wish to log events for debug purposes, create a file `config/analytics_event_debug.yml` containing an array of your event filters  under a `shared` key like:
+
+```yaml
+shared:
+  event_filters:
+    -
+      type: (create|update|delete)_entity
+      entity_table_name: course_options
+      key: id
+      value: 12345
```

Comment:
I find this example confusing because these key/value pairs actually refer to the contents of the nested `data` hash so the structure of this is not similar to the structure of an actual event. Would it be better to explicitly nest them under a `data` key? (I think this is what the code actually does in the spec!). Also, what would the process be for adding multiple key/value pairs? ([Event schema, for reference](https://github.com/DFE-Digital/dfe-analytics/blob/main/config/event-schema.json))

I wonder if this PR could be simplified by simply restricting to top-level string fields like `type`, `entity_table_name` etc, and then we could loop back and layer on `request_query` and `data` options?

Reply:
@duncanjbrown  - You are absolutely correct, key and value should be nested under data. The type key is also wrong! Sorry - my bad - I've updated now.

Multiple values for key/value pairs would require another filter. For example if we wanted to search for field `course_id` value `42` then
in the above example then we would add another filter to the array:

``` YAML
    -
      event_type: (create|update|delete)_entity
      entity_table_name: course_options
      data:
        key: course_id
        value: 42
```

The reason why I've done nesting like this is to keep the code generic. The code does not refer to any fields by name and does a match on top level fields and then nested fields by recursing over them.